### ISSUE TITLE
[xfreerdp] /monitors:2,0 constructs invalid monitor configuration

### ISSUE THREAD

**Issue by xBr0th3rx** [FreeRDP/FreeRDP#10910]
**Describe the bug**
I have 3 monitors in a row from left to right. All 3 monitors have the same resolution 1920x1080 and the coordinates are left (+0+0), center (+1920+0) and right (+3840+0). 
The main monitor is in the center.

I start Freerdp for the Windows session with 2 monitors left and center. Then I get an error message that the monitors are not next to each other or overlap.

Output from /list:monitor:
```
      * [0] 1920x1080   +1920+0
        [1] 1920x1080   +3840+0
        [2] 1920x1080   +0+0
```

**Expected behavior**
That I can start the Windows session via monitor 1 and 2.

**Application details**
* FreeRDP version (`xfreerdp /version`)
`This is FreeRDP version 3.9.1-dev0 (05801a162)`
* Command line used
`/opt/freerdp-nightly/bin/xfreerdp3 /d:XXX /u:XXX /v:XXX /multimon /monitors:2,0 /sound /clipboard /sec:tls /cert:ignore /dynamic-resolution +auto-reconnect /rfx /log-level:trace 2>&1 | tee log.txt`
* OS version connecting to (server side)
`Windows Terminal Server 2016`
* If available the log output from a run with `/log-level:trace 2>&1 | tee log.txt` 
[log.txt](https://github.com/user-attachments/files/18003395/log.txt)

**Environment (please complete the following information):**
 - OS: Linux
 - Version/Distribution: Debian 11
 - Architecture: amd64

**Comment by akallabeth** [comment]
@xBr0th3rx so this config works?
well, expected (the mapping is there but incomplete)

note: the `list:monitor` lists host coordinates, not the mapped windows coordinates (which are only available if you have all `/multimon /monitor:1,2,3` options applied)

**Comment by akallabeth** [comment]
@xBr0th3rx please.
the bug currently is that the mapping with `/monitor:0,2` in your initial example does not remap the used monitors in such a way that there are no gaps and the primary is at correct offset.

the check we have here is at a lower level where we send the configuration to the server which is already the remapped version that the server expects.
(and the whole point of this check was to bail out on configurations that were used but not correct so we get feedback to which cases we did not yet handle instead of - to us - cryptic issue reports later on)

**Comment by akallabeth** [comment]
@Voxnod you got it half right.
just as mentioned before, the mapping is incomplete, so the `0/0` mapping there is correct, but the other monitor is not mapped accordingly.

**Comment by smsisko** [comment]
@Voxnod I had a similar problem running under Wayland using Remmina and your code from `freerdp_settings_enforce_monitor_exists` pointed me in the right direction.

My monitors are in the following arrangement:
![image](https://github.com/user-attachments/assets/988fd871-5ce8-405f-b511-756106ed5d36)

When I used the following:

```c
BOOL freerdp_settings_enforce_monitor_exists(rdpSettings* settings)
{
    ...    
	else if (fullscreen || multimon)
	{
		/* not all platforms start primary monitor at 0/0, so enforce this to avoid issues with
		 * fullscreen mode */
		rdpMonitor* primaryMonitor = NULL;
		rdpMonitor* monitor =
		    freerdp_settings_get_pointer_array_writable(settings, FreeRDP_MonitorDefArray, 0);
		UINT32 x = 0;
		INT32 xOffset = 0;
		INT32 yOffset = 0;
		if (!monitor)
			return FALSE;
		for (; (x < count) && (primaryMonitor == NULL); ++x)
		{
			if (monitor[x].is_primary)
			{
				primaryMonitor = &monitor[x];
				xOffset = primaryMonitor->x;
				yOffset = primaryMonitor->y;
			}
		}
		if (primaryMonitor == NULL)
		{
			monitor->is_primary = TRUE;
			xOffset = monitor->x;
			yOffset = monitor->y;
		}
		for (x = 0; x < count; ++x)
		{
			monitor[x].x -= xOffset;
			monitor[x].y -= yOffset;
		}
	}
    ...    
}
```

This way I could keep the primary monitor needs to be at 0,0 requirement, and "moving" all the monitors relative to the primary.

**Comment by smsisko** [comment]
So, you're saying that, in my case, this should be done in Remmina ?

That doesn't feel right, forcing the primary monitor to be at 0,0 and all other monitors to be relative to it is an RDP requirements. It doesn't make any sense to me to force a client, and in this case a multiprotocol client, to implement protocol specific logic.

I don't understand how that relates to the platform, aren't those settings only used internally by the library ?
I wouldn't expect `freerdp_settings_*` to be platform specific.

To be clear, my change here is just to generalize the behaviour that was already coded for a single screen to be for all screen presents.

**Comment by smsisko** [comment]
@akallabeth first, thanks for your help, and I really don't want to be a pain or anything.

I think we have a misunderstanding.

Are you saying that what is already there, which we could describe as making sure the primary screen as it 0,0 should not be there ?

I'm asking because what I did could be described as making sure that all the monitors geometry is relative to the primary screen that needs to be at 0,0. In other words, the code does the same thing but for all the screens instead of just primary / first monitor.

To be clear, I understand your points, but I don't see how the code changes break any of those.

- Doesn't affect the current layout (unlike the current code)
- Again, not changing the layout, just moving the layout so that the primary screen is at 0,0, just like before
- Those changes should have no impact on the platform, unless they had before, since I'm just doing the same thing but for the whole layout instead of part of it.

**Comment by akallabeth** [comment]
@smsisko yes, that is more or less something that should not be there and just a check if the requirements are met. (but then even less configurations would work with current state)

at that place there should already be a correct monitor layout set and that should not require manual intervention, but there is a lot to do and a lot of configurations to consider.

as for no impact, every mouse move/click will break if the layouts move as the coordinates transmitted will be different

**Comment by smsisko** [comment]
@akallabeth I will try to find out where that configuration is initially set up and see how that could be changed initially.

But right now, the mouse works fine.

With those changes, I went from a configuration that couldn't even start (I had the same overlap error that Voxnod had) to a system that is almost completely usable (right now the only thing still missing is the floatbar location that is relative to the whole geometry in Y instead of the primary screen).

**Comment by akallabeth** [comment]
@smsisko for `X11` the changes are most likely fine.
`FreeRDP` is running on a variety of hard and software though (also some clients that use windows as monitors) so adding this to the core library is a mistake. (just like the recently deprecated `freerdp_keyboard_get_rdp_scancode_from_x11_keycode` which did work for some cases but was completely broken for others as it was just in the wrong place)

we´re adding low level checks for compliance and this is a case where the clients (even our own) did just blatantly ignore the spec.

**Comment by smsisko** [comment]
My current environment is Wayland with Gnome (in Fedora 41). 
I am connecting using Remmina, so I am looking in that plugin to see if I can make equivalent changes.
I tried connecting the SDL clients, but I end up with some security problems trying to connect to the Windows machine, so that is the only way I have tested to date.

**Comment by xBr0th3rx** [comment]
Hey @akallabeth, 
I have tested this and now I can start it without my main monitor having to be at 0,0 on the left and I don't get an overlap message.

Thank you for your great work.
